### ISSUE TITLE
Cannot build IOS appld: "ld: library not found for -l" plugin libraries

### ISSUE THREAD

**Issue by rlee1990** [flutter/flutter#24067]
Created a new flutter project using objective c not swift and receive the following error trying to build. This was not an issue last night no change in code.
```
Launching lib/main.dart on iPhone XR in debug mode...
Running pod install...
Starting Xcode build...
Xcode build done.                                            2.8s
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **


Xcode's output:
↳
    ld: warning: directory not found for option '-L/Volumes/drive/Flutter_Apps/obcjc/flutter_crushn/build/ios/Debug-iphonesimulator/BoringSSL'
    ld: warning: directory not found for option '-L/Volumes/drive/Flutter_Apps/obcjc/flutter_crushn/build/ios/Debug-iphonesimulator/FirebaseAuth'
    ld: warning: directory not found for option '-L/Volumes/drive/Flutter_Apps/obcjc/flutter_crushn/build/ios/Debug-iphonesimulator/FirebaseDatabase'
    ld: warning: directory not found for option '-L/Volumes/drive/Flutter_Apps/obcjc/flutter_crushn/build/ios/Debug-iphonesimulator/FirebaseFirestore'
    ld: warning: directory not found for option '-L/Volumes/drive/Flutter_Apps/obcjc/flutter_crushn/build/ios/Debug-iphonesimulator/FirebaseMessaging'
    ld: warning: directory not found for option '-L/Volumes/drive/Flutter_Apps/obcjc/flutter_crushn/build/ios/Debug-iphonesimulator/FirebaseStorage'
    ld: warning: directory not found for option '-L/Volumes/drive/Flutter_Apps/obcjc/flutter_crushn/build/ios/Debug-iphonesimulator/GoogleAPIClientForREST'
    ld: warning: directory not found for option '-L/Volumes/drive/Flutter_Apps/obcjc/flutter_crushn/build/ios/Debug-iphonesimulator/GoogleToolboxForMac'
    ld: warning: directory not found for option '-L/Volumes/drive/Flutter_Apps/obcjc/flutter_crushn/build/ios/Debug-iphonesimulator/camera'
    ld: warning: directory not found for option '-L/Volumes/drive/Flutter_Apps/obcjc/flutter_crushn/build/ios/Debug-iphonesimulator/cloud_firestore'
    ld: warning: directory not found for option '-L/Volumes/drive/Flutter_Apps/obcjc/flutter_crushn/build/ios/Debug-iphonesimulator/firebase_admob'
    ld: warning: directory not found for option '-L/Volumes/drive/Flutter_Apps/obcjc/flutter_crushn/build/ios/Debug-iphonesimulator/firebase_analytics'
    ld: warning: directory not found for option '-L/Volumes/drive/Flutter_Apps/obcjc/flutter_crushn/build/ios/Debug-iphonesimulator/firebase_auth'
    ld: warning: directory not found for option '-L/Volumes/drive/Flutter_Apps/obcjc/flutter_crushn/build/ios/Debug-iphonesimulator/firebase_core'
    ld: warning: directory not found for option '-L/Volumes/drive/Flutter_Apps/obcjc/flutter_crushn/build/ios/Debug-iphonesimulator/firebase_database'
    ld: warning: directory not found for option '-L/Volumes/drive/Flutter_Apps/obcjc/flutter_crushn/build/ios/Debug-iphonesimulator/firebase_messaging'
    ld: warning: directory not found for option '-L/Volumes/drive/Flutter_Apps/obcjc/flutter_crushn/build/ios/Debug-iphonesimulator/firebase_storage'
    ld: warning: directory not found for option '-L/Volumes/drive/Flutter_Apps/obcjc/flutter_crushn/build/ios/Debug-iphonesimulator/flutter_inapp_purchase'
    ld: warning: directory not found for option '-L/Volumes/drive/Flutter_Apps/obcjc/flutter_crushn/build/ios/Debug-iphonesimulator/flutter_native_image'
    ld: warning: directory not found for option '-L/Volumes/drive/Flutter_Apps/obcjc/flutter_crushn/build/ios/Debug-iphonesimulator/flutter_statusbar_manager'
    ld: warning: directory not found for option '-L/Volumes/drive/Flutter_Apps/obcjc/flutter_crushn/build/ios/Debug-iphonesimulator/flutter_webview_plugin'
    ld: warning: directory not found for option '-L/Volumes/drive/Flutter_Apps/obcjc/flutter_crushn/build/ios/Debug-iphonesimulator/gRPC-C++'
    ld: warning: directory not found for option '-L/Volumes/drive/Flutter_Apps/obcjc/flutter_crushn/build/ios/Debug-iphonesimulator/gRPC-Core'
    ld: warning: directory not found for option '-L/Volumes/drive/Flutter_Apps/obcjc/flutter_crushn/build/ios/Debug-iphonesimulator/gRPC-ProtoRPC'
    ld: warning: directory not found for option '-L/Volumes/drive/Flutter_Apps/obcjc/flutter_crushn/build/ios/Debug-iphonesimulator/geocoder'
    ld: warning: directory not found for option '-L/Volumes/drive/Flutter_Apps/obcjc/flutter_crushn/build/ios/Debug-iphonesimulator/image_picker'
    ld: warning: directory not found for option '-L/Volumes/drive/Flutter_Apps/obcjc/flutter_crushn/build/ios/Debug-iphonesimulator/intro_slider'
    ld: warning: directory not found for option '-L/Volumes/drive/Flutter_Apps/obcjc/flutter_crushn/build/ios/Debug-iphonesimulator/leveldb-library'
    ld: warning: directory not found for option '-L/Volumes/drive/Flutter_Apps/obcjc/flutter_crushn/build/ios/Debug-iphonesimulator/location'
    ld: warning: directory not found for option '-L/Volumes/drive/Flutter_Apps/obcjc/flutter_crushn/build/ios/Debug-iphonesimulator/mlkit'
    ld: warning: directory not found for option '-L/Volumes/drive/Flutter_Apps/obcjc/flutter_crushn/build/ios/Debug-iphonesimulator/native_device_orientation'
    ld: warning: directory not found for option '-L/Volumes/drive/Flutter_Apps/obcjc/flutter_crushn/build/ios/Debug-iphonesimulator/onesignal'
    ld: warning: directory not found for option '-L/Volumes/drive/Flutter_Apps/obcjc/flutter_crushn/build/ios/Debug-iphonesimulator/path_provider'
    ld: warning: directory not found for option '-L/Volumes/drive/Flutter_Apps/obcjc/flutter_crushn/build/ios/Debug-iphonesimulator/shared_preferences'
    ld: warning: directory not found for option '-L/Volumes/drive/Flutter_Apps/obcjc/flutter_crushn/build/ios/Debug-iphonesimulator/video_player'
    ld: library not found for -lFirebaseAuth
    clang: error: linker command failed with exit code 1 (use -v to see invocation)

Could not build the application for the simulator.
Error launching application on iPhone XR.`
```

Ran `flutter clean` issue remains.

**Comment by zoechi** [comment]
Please add the output of `flutter doctor -v`.

**Comment by rlee1990** [comment]
`[✓] Flutter (Channel master, v0.10.3-pre.128, on Mac OS X 10.13.6 17G65, locale en-US)
    • Flutter version 0.10.3-pre.128 at /Users/rickeylee/Desktop/flutter
    • Framework revision 9abce96e8e (2 days ago), 2018-11-07 12:20:27 -0500
    • Engine revision e64af7510e
    • Dart version 2.1.0 (build 2.1.0-dev.8.0 bf26f760b1)

[✓] Android toolchain - develop for Android devices (Android SDK 28.0.3)
    • Android SDK at /Users/rickeylee/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b06)
    • All Android licenses accepted.

[✓] iOS toolchain - develop for iOS devices (Xcode 10.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 10.1, Build version 10B61
    • ios-deploy 2.0.0
    • CocoaPods version 1.5.3

[✓] Android Studio (version 3.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 30.0.1
    • Dart plugin version 181.5656
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b06)

[✓] VS Code (version 1.28.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 2.20.0

[✓] Connected device (1 available)
    • SM N950U • ce061716d1923703017e • android-arm64 • Android 8.0.0 (API 26)

• No issues found!
`

**Comment by rlee1990** [comment]
@zoechi

**Comment by rlee1990** [comment]
Any updates?

**Comment by strokine** [comment]
I have exactly the same problem. The list of the directories is shorter and sometimes changes. I also have `image_picker` and `onesignal`. I started to see this error after I've added `onesignal`. But originally everything worked, but at some point, it stopped. Nothing dramatic was done, same flutter version, just some versions of some other libraries upgraded maybe.
Struggling with it for a few days already.
